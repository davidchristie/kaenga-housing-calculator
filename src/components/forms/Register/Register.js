import React, { Component, PropTypes } from 'react'
import { Button } from 'react-bootstrap'
import { Field } from 'redux-form'

import TextInput from '../../controls/TextInput'

class Register extends Component {
  render () {
    const { handleSubmit, invalid, register } = this.props
    return (
      <form onSubmit={handleSubmit(register)}>
        <Field
          component={TextInput}
          label='First name'
          name='firstName'
        />
        <Field
          component={TextInput}
          label='Last name'
          name='lastName'
        />
        <Field
          component={TextInput}
          label='Phone number'
          name='phone'
          type='tel'
        />
        <Field
          component={TextInput}
          label='Email'
          name='email'
          type='email'
        />
        <p>
          Click submit to find out more about your ideal home. We’ll also show you what it might look like.
        </p>
        <p>
          Even better, we’ll take into account your preferences when designing future projects and give you the opportunity to buy before releasing the project to the public.
        </p>
        <Button
          bsStyle='success'
          disabled={invalid}
          type='submit'
        >
          Submit
        </Button>
      </form>
    )
  }
}

Register.defaultProps = {
  handleSubmit: () => {}
}

Register.propTypes = {
  handleSubmit: PropTypes.func
}

export default Register
