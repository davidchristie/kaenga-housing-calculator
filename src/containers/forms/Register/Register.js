import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'

import { register } from '../../../actions/register'
import Register from '../../../components/forms/Register'

const validate = values => {
  const errors = {}
  if (!values.firstName) {
    errors.firstName = 'Required'
  } else if (values.firstName.length > 15) {
    errors.firstName = 'Must be 15 characters or less'
  }
  if (!values.lastName) {
    errors.lastName = 'Required'
  } else if (values.lastName.length > 15) {
    errors.lastName = 'Must be 15 characters or less'
  }
  if (!values.phone) {
    errors.phone = 'Required'
  }
  if (!values.email) {
    errors.email = 'Required'
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address'
  }
  return errors
}

export default connect(
  state => {
    const register = state.form.registry || {}
    return {
      values: register.values
    }
  },
  dispatch => {
    return {
      register: values => dispatch(register(values))
    }
  }
)(
  reduxForm({
    destroyOnUnmount: false,
    form: 'register',
    initialValues: {
      email: '',
      firstName: '',
      lastName: '',
      phone: ''
    },
    validate
  })(Register)
)
