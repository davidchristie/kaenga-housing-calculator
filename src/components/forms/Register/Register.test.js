import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import Register from './Register'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<Register />)).toMatchSnapshot()
})
