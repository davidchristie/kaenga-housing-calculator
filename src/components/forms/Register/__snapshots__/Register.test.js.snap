exports[`test matches snapshot 1`] = `
<form
  onSubmit={undefined}>
  <Field
    component={[Function]}
    label="First name"
    name="firstName" />
  <Field
    component={[Function]}
    label="Last name"
    name="lastName" />
  <Field
    component={[Function]}
    label="Email"
    name="email"
    type="email" />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    type="submit">
    Submit
  </Button>
</form>
`;
